addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java

```diff
@@ -696,20 +714,31 @@ private void updateNotificationTimerState() {
 
     public void updateState(AccountHandler accountHandler, @Nullable Device device,
             @Nullable BluetoothState bluetoothState, @Nullable DeviceNotificationState deviceNotificationState,
-            @Nullable AscendingAlarmModel ascendingAlarmModel) {
+            @Nullable AscendingAlarmModel ascendingAlarmModel, @Nullable JsonPlaylists playlists,
+            JsonNotificationSound @Nullable [] alarmSounds, @Nullable List<JsonMusicProvider> musicProviders) {
```

Comment:
Missing `@Nullable` in front of `JsonNotificationSound`?

Reply:
No sure about the using of Nullable, but yes the alarmSounds can be null, so I have added the @Nullable in front of the type. What is the meaning of @Nullable before the brackets? Does it mean, that array entries can be null?

Reply:
To be honest, I had to Google the usage with arrays too. Here is what I got: http://help.eclipse.org/luna/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_null_type_annotations.htm

## Syntax

For two constructs the JLS introduces a syntactic change:

| Declaration Annotations (Java 7 or below)	|Type Annotation (Java 8) |
|-|-|
| @NonNull String[] |	String @NonNull[] |
| @NonNull java.lang.String |	java.lang.@NonNull String |

In both cases the new syntax has been introduced to provide more options.

For arrays a type annotation before the leaf element type will now denote an array whose individual cells have the given nullness - here: cells cannot be null. In Java 7 and below the same syntax expressed a property of the corresponding variable and hence captured the nullness of the array itself. To express the same using Java-8 type annotations, viz. that the array itself can or cannot be null, the type annotation is placed before the square brackets denoting the array dimensions. This implies that the old syntax is still valid, but its meaning has changed:
```
    // annotated leaf type:
    @NonNull Object [] o1;
    o1 = null;          // OK
    o1 = new Object[1];
    o1[0] = null;       // NOT OK
    ...
    // annotated array type:
    Object @NonNull[] o2;
    o2 = null;          // NOT OK
    o2 = new Object[1];
    o2[0] = null;       // OK
    ...
    // multi-dimensional array:
    Object @NonNull[] @Nullable[] o3;
    o3 = null;          // NOT OK, outer array is nonnull
    o3 = new Object[1] @Nullable[];
    o3[0] = null;       // OK, inner array is nullable
```
Unfortunately, checking proper initialization of an array with nonnull content is beyond the capabilities of JDT's static analysis.

Reply:
Thank you!

Reply:
Alright then. Do you think we have to change something for the `JsonNotificationSound[] alarmSounds` array? Does it contain `null` values?